classes/Site/Popups.php

```diff
@@ -42,6 +42,9 @@ public static function init() {
 		// Preload the $loaded query.
 		add_action( 'init', array( __CLASS__, 'get_loaded_popups' ) );
 
+		// Check content for popups.
+		add_filter( 'the_content', array( __CLASS__, 'check_content_for_popups' ) );
```

Comment:
Doing this via `the_content` filter will be very ineficcient. Specifically you would be adding our preg search every time they call for content.

We should only be doing this once, and likely not during that filter but explicitly after `global $post` is set or the main query results are available for looping over.

Reply:
@danieliser I went back-n-forth with this a bit. Because the_content could also be content within their sidebar where they want a popup to open as well as other areas. If we do limit it to the main page content, the docs suggest the conditional I included in the issue, `is_singular() && in_the_loop() && is_main_query() ` which would ensure this is only run once for the main content.

Reply:
Your call, but we may run into issues down the line either way.